London | 25-SDC-July | Andrei Filippov | Sprint 4 | Implement shell tools in Python

Droid-An opened the pull request:
## Learners, PR Template

Self checklist

- [x] I have committed my files one by one, on purpose, and for a reason
- [x] I have titled my PR with Region | Cohort | FirstName LastName | Sprint | Assignment Title
- [x] I have tested my changes
- [x] My changes follow the [style guide](https://curriculum.codeyourfuture.io/guides/reviewing/style-guide/)
- [x] My changes meet the [requirements](./README.md) of this task

## Changelist

Implemented shell tools in Python 

## Questions

what formatter to use for python?

Droid-An committed:
wip: done -1 flag

Droid-An committed:
done ls

Droid-An committed:
removed unnecessary  comments and created wc.py file

Droid-An committed:
done wc

Droid-An committed:
done cat

LonMcGregor reviewed (commented):
Good start on these implementations - I've left a comment on each file with a pointer to show you where you could improve them.

For your question about formatters what do you mean - formatting your code or formatting strings?
* You can use any code formatter (or none at all), as long as you are consistent with the code you commit
* Python has a few ways to do string formatting, the newest versions of python have something called "f-strings" that are very powerful, but might take a lot of time to learn fully

Droid-An committed:
done changes according to review

LonMcGregor reviewed (commented):
* For cat, try running the original cat program with the `-n`argument, with multiple files using a wildcard, and then run your own. Can you see any differences?
* Can you see any issues with the alignment of columns if you run ẁc`with multiple files using a wildcard?

LonMcGregor commented on implement-shell-tools/ls/ls.py:
```diff
@@ -0,0 +1,46 @@
+path_proceeding(args.path)
+
+
+
```
Thanks for fixing this

Droid-An committed:
fixed cat

Droid-An committed:
fixed wc

Droid-An committed:
done changes according to review

Droid-An committed:
added right-justify to output

Droid-An commented:
- I saw difference in spacing, so I added more space to output. Also I aligned numbers to the right and padded with spaces on the left.
- yes, numbers in output were not right-aligned; they were left-aligned with inconsistent spacing. I fixed it by applying `rjust` method before printing

LonMcGregor reviewed (commented):
Good work, I think you are almost there with this implementation.

Droid-An committed:
Applied review changes
